app/src/main/java/com/android/sample/ui/filter/FilterPage.kt

```diff
@@ -3,63 +3,46 @@ package com.android.sample.ui.filter
-import androidx.compose.foundation.layout.FlowRow
 import androidx.compose.foundation.layout.PaddingValues
 import androidx.compose.foundation.layout.Row
-import androidx.compose.foundation.layout.Spacer
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.fillMaxWidth
-import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.wrapContentSize
 import androidx.compose.foundation.rememberScrollState
 import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.foundation.verticalScroll
 import androidx.compose.material3.Button
 import androidx.compose.material3.ButtonDefaults
-import androidx.compose.material3.Checkbox
 import androidx.compose.material3.MaterialTheme
-import androidx.compose.material3.RangeSlider
-import androidx.compose.material3.SliderDefaults
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateMapOf
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
-import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
 import com.android.sample.R
 import com.android.sample.model.filter.Difficulty
 import com.android.sample.model.filter.FilterPageViewModel
-import com.android.sample.model.filter.FloatRange
-import com.android.sample.resources.C.Dimension.PADDING_16
 import com.android.sample.resources.C.Dimension.PADDING_4
 import com.android.sample.resources.C.Dimension.PADDING_8
 import com.android.sample.resources.C.Dimension.SwipePage.BUTTON_ELEVATION
 import com.android.sample.resources.C.Dimension.SwipePage.BUTTON_RADIUS
 import com.android.sample.resources.C.Tag.Filter.UNINITIALIZED_BORN_VALUE
-import com.android.sample.resources.C.Tag.FilterPage.MAX_ITEM_IN_ROW
-import com.android.sample.resources.C.Tag.FilterPage.SLIDER_COLOR_ACTIVE
-import com.android.sample.resources.C.Tag.FilterPage.SLIDER_COLOR_INACTIVE
 import com.android.sample.resources.C.Tag.FilterPage.TIME_RANGE_MAX
 import com.android.sample.resources.C.Tag.FilterPage.TIME_RANGE_MIN
-import com.android.sample.resources.C.Tag.MINUTES_PER_HOUR
 import com.android.sample.resources.C.TestTag.FilterPage.TEST_TAG_CATEGORY
 import com.android.sample.resources.C.TestTag.FilterPage.TEST_TAG_DIFFICULTY
 import com.android.sample.resources.C.TestTag.FilterPage.TEST_TAG_TIME_RANGE_SLIDER
 import com.android.sample.resources.C.TestTag.SwipePage.VIEW_RECIPE_BUTTON
 import com.android.sample.ui.navigation.NavigationActions
 import com.android.sample.ui.navigation.Screen
+import com.android.sample.ui.utils.CheckboxGroup
```

Comment:
- **Button Swap**: Swapping the `Reset` and `Apply` buttons based on user feedback is a solid move! It’s great to see user input being incorporated into the design. 👌

- **Code Extraction**: Moving utility functions like `ValueRangeSlider` and `CheckboxGroup`  to `ValueControl.kt` is definitely the right call. It makes `FilterPage.kt` lighter and improves code organization. Great for maintainability and reusability! 💡

Reply:
Thanks